tests/unit/providers/agents/meta_reference/test_openai_responses.py

```diff
@@ -0,0 +1,202 @@
+    OpenAIChoice,
+    OpenAIUserMessageParam,
+)
+from llama_stack.apis.tools.tools import Tool, ToolGroups, ToolInvocationResult, ToolParameter, ToolRuntime
+from llama_stack.providers.inline.agents.meta_reference.openai_responses import (
+    OpenAIResponsesImpl,
+)
+from llama_stack.providers.utils.kvstore import KVStore
+
+
+@pytest.fixture
+def mock_kvstore():
+    kvstore = AsyncMock(spec=KVStore)
+    return kvstore
+
+
+@pytest.fixture
+def mock_inference_api():
+    inference_api = AsyncMock()
+    return inference_api
+
+
+@pytest.fixture
+def mock_tool_groups_api():
+    tool_groups_api = AsyncMock(spec=ToolGroups)
+    return tool_groups_api
+
+
+@pytest.fixture
+def mock_tool_runtime_api():
+    tool_runtime_api = AsyncMock(spec=ToolRuntime)
+    return tool_runtime_api
+
+
+@pytest.fixture
+def openai_responses_impl(mock_kvstore, mock_inference_api, mock_tool_groups_api, mock_tool_runtime_api):
+    return OpenAIResponsesImpl(
+        persistence_store=mock_kvstore,
+        inference_api=mock_inference_api,
+        tool_groups_api=mock_tool_groups_api,
+        tool_runtime_api=mock_tool_runtime_api,
+    )
+
+
+@pytest.mark.asyncio
+async def test_create_openai_response_with_string_input(openai_responses_impl, mock_inference_api):
+    """Test creating an OpenAI response with a simple string input."""
+    # Setup
+    input_text = "Hello, world!"
+    model = "meta-llama/Llama-3.1-8B-Instruct"
+
+    mock_chat_completion = OpenAIChatCompletion(
+        id="chat-completion-123",
+        choices=[
+            OpenAIChoice(
+                message=OpenAIAssistantMessageParam(content="Hello! How can I help you?"),
+                finish_reason="stop",
+                index=0,
+            )
+        ],
+        created=1234567890,
+        model=model,
+    )
+    mock_inference_api.openai_chat_completion.return_value = mock_chat_completion
+
+    # Execute
+    result = await openai_responses_impl.create_openai_response(
+        input=input_text,
+        model=model,
+        temperature=0.1,
+    )
+
+    # Verify
+    mock_inference_api.openai_chat_completion.assert_called_once_with(
+        model=model,
+        messages=[OpenAIUserMessageParam(role="user", content="Hello, world!", name=None)],
+        tools=None,
+        stream=False,
+        temperature=0.1,
+    )
+    openai_responses_impl.persistence_store.set.assert_called_once()
+    assert result.model == model
+    assert len(result.output) == 1
+    assert isinstance(result.output[0], OpenAIResponseOutputMessage)
+    assert result.output[0].content[0].text == "Hello! How can I help you?"
+
+
+@pytest.mark.asyncio
+async def test_create_openai_response_with_string_input_with_tools(openai_responses_impl, mock_inference_api):
+    """Test creating an OpenAI response with a simple string input and tools."""
+    # Setup
+    input_text = "What was the score of todays game?"
+    model = "meta-llama/Llama-3.1-8B-Instruct"
+
+    mock_chat_completions = [
```

Comment:
Thank you so much for bootstrapping unit tests here, as there are other pieces of the Responses implementation (mostly in translation from Responses -> Chat Completion, Chat Completion Result -> Response Result, conversation state, and so on) that I need to add tests for and fix some unimplemented / bugs I've already identified. There is quite a bit of setup code in some of these tests, but as more tests are written we can figure out a way to factor out as much as possible to simplify the mock and setup required.

Reply:
Ya there is scope to clean up a lot their, I think I was happy to eventually get them working...
I'm hoping to add some more as I add support for more params, I can clean them up then.

Reply:
For sure - as we add more tests it should be easier to identify the common patterns and extract bits out. I'm impressed you wired these up to begin with!